src/spyglass/utils/dj_merge_tables.py

```diff
@@ -0,0 +1,425 @@
+                        )
+                    else:
+                        raise ValueError(
+                            "Mutual Exclusivity Error! Entry exists in more "
+                            + f"than one table - Entry: {row}"
+                        )
+
+            if not key:
+                raise ValueError(
+                    "Non-existing entry in any of the parent tables - Entry: "
+                    + f"{row}"
+                )
+
+        # 1. nullcontext() allows use within `make` but decreases reliability
+        # 2. cls.connection.transaction is more relaiable but throws errors if
+        # used within another transaction, i.e. in `make`
+
+        with nullcontext():  # TODO: ensure this block within transaction
+            super().insert(cls(), master_entries, **kwargs)
+            for part, part_entries in parts_entries.items():
+                part.insert(part_entries, **kwargs)
+
+    @classmethod
+    def insert(cls, rows: list, **kwargs):
+        """Insert rows into merge table
+
+        Ensuring db integrity and mutual exclusivity
+
+        Parameters
+        ---------
+        rows: List[dict]
+            An iterable where an element is a dictionary.
+
+        Raises
+        ------
+        TypeError
+            If rows is not a list of dicts
+        ValueError
+            If entry already exists, mutual exclusivity errors
+            If data doesn't exist in part parents, integrity error
+        """
+        cls._merge_insert(rows, **kwargs)
+
+    @classmethod
+    def merge_view(cls, restriction: str = True):
+        """Prints merged view, including null entries for unique columns.
+
+        Parameters
+        ---------
+        restriction: str, optional
+            Restriction to apply to the merged view
+        """
+
+        # If we overwrite `preview`, we then encounter issues with operators
+        # getting passed a `Union`, which doesn't have a method we can
+        # intercept to manage master/parts
+
+        return pprint(cls._merge_repr(restriction=restriction))
+
+    @classmethod
+    def merge_html(cls, restriction: str = True):
+        """Displays HTML in notebooks."""
+
+        return HTML(repr_html(cls._merge_repr(restriction=restriction)))
+
+    @classmethod
+    def merge_restrict(cls, restriction: str = True) -> dj.U:
+        """Given a restriction string, return a merged view with restriction applied.
+
+        Example
+        -------
+            >>> MergeTable.merge_restrict("field = 1")
+
+        Parameters
+        ----------
+        restriction: str
+            Restriction one would appy if `merge_view` was a real table.
+
+        Returns
+        -------
+        datajoint.Union
+            Merged view with restriction applied.
+        """
+        return cls._merge_repr(restriction=restriction)
+
+    @classmethod
+    def merge_delete(cls, restriction: str = True, **kwargs):
```

Comment:
I'm not sure about the intended usage for this, but would it make sense to have it also delete from the part table? Or is it assumed the user will pass `"force_parts=True"` as a keyword arg? I may also be unclear on the delete process when deleting from the parent table first..

Reply:
This deletes the master primary key, which cascades down to the part. `force_parts` overrides the orphaned-master protection on parts when cascading down from a different parent, but not needed in this case